pallets/settlement/src/lib.rs

```diff
@@ -1190,24 +1252,37 @@ impl<T: Config> Module<T> {
 
     fn ensure_instruction_validity(
         id: InstructionId,
+        is_execute: bool,
     ) -> Result<Instruction<T::Moment, T::BlockNumber>, DispatchError> {
         let details = Self::instruction_details(id);
         ensure!(
             details.status != InstructionStatus::Unknown,
             Error::<T>::UnknownInstruction
         );
-        if let (Some(trade_date), Some(value_date)) = (details.trade_date, details.value_date) {
-            ensure!(
-                value_date >= trade_date,
-                Error::<T>::InstructionDatesInvalid
-            );
-        }
-        if let SettlementType::SettleOnBlock(block_number) = details.settlement_type {
-            ensure!(
-                block_number > System::<T>::block_number(),
-                Error::<T>::InstructionSettleBlockPassed
-            );
+
+        match details.settlement_type {
+            SettlementType::SettleOnBlock(block_number) => {
+                if is_execute {
+                    ensure!(
+                        block_number > System::<T>::block_number(),
+                        Error::<T>::InstructionSettleBlockPassed
+                    );
+                } else {
+                    ensure!(
+                        block_number <= System::<T>::block_number(),
+                        Error::<T>::InstructionSettleBlockPassed
+                    );
+                }
```

Comment:
These two checks are backwards.  Also please add comments explaining the checks.